doc/source/reference/arrays.nditer.rst

```diff
@@ -220,20 +220,37 @@ produce identical results to the ones in the previous section.
     ...     print("%d <%d>" % (it[0], it.index), end=' ')
     ...     it.iternext()
```

Comment:
It's nasty, but this would make the output go away:
```suggestion
    ...     it.iternext()
    ...     None  # needed to make the docs build properly :(
```

Reply:
Why not `v = it.iternext()`

Reply:
Well, the output that I put in the PR corresponds to the REPL output, that differs from the IPython output. It is a matter of choice between the two.

Reply:
I think we could make both the repl and ipython have the same output if we change line 221 to be `v = it.iternext()` (and line 233 too)

Reply:
I agree. However, it remains hacky: the example is modified for the purpose of a clean doctest but the modification does not reflect a programming necessity or a good practice. That being said, you are in a much better position to make the choice for this matter of style, let me know your decision.

Reply:
I think it is easier to hide this detail in an unused value than to try to explain why the loop is printing booleans. Good practice would be to use the `for` loop anyway, I am not a big fan of looping over an iterator with `while`.

Reply:
Perhaps `is_not_finished = ...` instead?

Reply:
> , I am not a big fan of looping over an iterator with while.

Even that would be a better example, since `next` is a lot more useful than our `iternext`.

Reply:
So, for this PR, we keep the `while` and I add `is_not_finished = `?

Reply:
yes please